### ISSUE TITLE
New standardised test results file format?

### ISSUE THREAD

**Issue by thomhurst** [microsoft/testfx#3214]
There's support for the trx file format in the new test runner, but that's been around for quite some time now.

I'd argue it's got some flaws such as:
- Lack of extensibility such as adding custom data.
- Exception details lack the actual exception type? Providing only message and stacktrace. The type itself seems pretty important?
- It's XML - I know it's not necessarily a file designed to be human readable, but with how JSON has progressed, it could be a better choice for a new test format.

**Comment by Evangelink** [comment]
YES!!! Hum sorry ;) It's on our plan but we had to adjust some priorities and got some pushback. I was planning on looking at other languages to see what they are doing and if their format could either be reused or could be used as inspiration.

**Comment by Evangelink** [comment]
Closing as won't fix because this is unlikely to get prioritized and we will get pushback from leadership about introducing something new.
